File: database/models/Homework.ts
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck
import { Model } from '@nozbe/watermelondb';
import { field} from "@nozbe/watermelondb/decorators";

export default class Homework extends Model {
  static table = 'homework';

  static associations = {
    subjects: { type: 'belongs_to', key: 'subjectId' },
  };

  @field('createdByAccount') createdByAccount: string;
	@field('kidName') kidName: string;
  @field('homeworkId') homeworkId: string;
  @field('subject') subject: string;
  @field('content') content: string;
  @field('dueDate') dueDate: number;
  @field('isDone') isDone: boolean;
  @field('returnFormat') returnFormat: number;
  @field('attachments') attachments: string;
  @field('evaluation') evaluation: boolean;
  @field('custom') custom: boolean;
}

File: targets/widget/Database/Models/Homework.swift
//
//  Homework.swift
//  Papillon
//
//  Created by Rémy Godet on 01/02/2026.
//

struct HomeworkModel {
  let createdByAccount: String;
  let kidName: String;
  let homeworkId: String;
  let subject: String;
  let content: String;
  let dueDate: Int64;
  let isDone: Bool;
  let returnFormat: Int;
  let attachments: String;
  let evaluation: Bool;
  let custom: Bool;
}
